src/cdk/datetime/dateadapter/date-adapter.ts

```diff
@@ -10,24 +10,16 @@ import {inject, InjectionToken, LOCALE_ID} from '@angular/core';
 import {Observable, Subject} from 'rxjs';
 
 /** InjectionToken for datepicker that can be used to override default locale code. */
-export const MAT_DATE_LOCALE = new InjectionToken<string>('MAT_DATE_LOCALE', {
+export const CDK_DATE_LOCALE = new InjectionToken<string>('CDK_DATE_LOCALE', {
   providedIn: 'root',
-  factory: MAT_DATE_LOCALE_FACTORY,
+  factory: CDK_DATE_LOCALE_FACTORY,
 });
 
 /** @docs-private */
-export function MAT_DATE_LOCALE_FACTORY(): string {
+export function CDK_DATE_LOCALE_FACTORY(): string {
   return inject(LOCALE_ID);
 }
 
-/**
- * No longer needed since MAT_DATE_LOCALE has been changed to a scoped injectable.
- * If you are importing and providing this in your code you can simply remove it.
- * @deprecated
- * @deletion-target 7.0.0
- */
-export const MAT_DATE_LOCALE_PROVIDER = {provide: MAT_DATE_LOCALE, useExisting: LOCALE_ID};
```

Comment:
technically we can't remove this until the 7.0 release, though this PR will not go out sooner than that, so this is probably fine

Reply:
Probably around November, we aim for ~6 months between major versions

Reply:
Do document this as a BREAKING CHANGE in the description though